### DISCUSSION TITLE
LedPanelWiFi проблемма с файлами на SD

### DISCUSSION THREAD

**Discussion by Shark2805** [vvip-68/LedPanelWiFi#23] [Вопросы и ответы]
Доброго всем дня. Помогите пожалуйста с проблемой. Файлы с ошибками. Не знаю куда копать. Пробовал и свои файлы и скачанные с интернета, у людей рабочие.
Инициализация SD-карты...
SD-карта подключена.
Поиск файлов эффектов Jinx!..
Папка с эффектами /18x23 обнаружена.
Загрузка списка файлов с эффектами...
   1                116.02 К               ERR - ошибка формата
   10               155.86 К               ERR - ошибка формата
   11               150.00 К               ERR - ошибка формата
   12               163.77 К               ERR - ошибка формата
   13               155.27 К               ERR - ошибка формата
   14               169.04 К               ERR - ошибка формата
   15               176.37 К               ERR - ошибка формата
   2                 77.34 К               ERR - ошибка формата
   3                156.15 К               ERR - ошибка формата
   4                175.78 К               ERR - ошибка формата
   5                141.21 К               ERR - ошибка формата
   6                186.04 К               ERR - ошибка формата
   7                157.32 К               ERR - ошибка формата
   8                182.81 К               ERR - ошибка формата
   9                180.76 К               ERR - ошибка формата
Доступно 15 файлов эффектов.

а это пишет при выборе эффектов с SD

Режим: SD-Карта
Загрузка файла эффекта: '/18x23/.out' -> ошибка

**Comment by vvip-68** [comment]
1. Оно так пишет, когда размер матрицы вашей гирлянды не соответствует размеру матрице в кадре файла эффектов.
2. Судя по надписи - имя файла пустое, только расширение. Как так у вас получается? Что за файлы то? Один из тех, которые с неправильным размером кадра?

**Reply by vvip-68** [reply]
По пункту 2. Это ошибка в скетче потом исправлю. При проверке списка найденных эффектов, те которые с ошибкой формата - у них обнуляется имя файла. Однако при открытии забыл сделать проверку - если имя эффекта сброшено (пустое) то его не использовать. Отсюда попытка открыть файл с пустым именем.

Ну то есть - уберете все файлы с неправильным размером кадра, оставите только с правильным - обе ошибки уйдут и эффект будет работать правильно. 

Правильный размер кадра под вашу гирлянду, очевидно, 18x23. Какой размер в файле можно посмотреть утилитой JinxFramer. Не помню точно, но возможно этой же утилитой можно подрезать до нужного размера, если размер исходного кадра больше вашего. А вот если меньше - уже, кажется, ничего с этим не сделаешь...

Этой же утилитой можно подправить угол подключения и направление из угла для ролика. С матрицами неправильной структуры (составными матрицами с индексными файлами) эта утилита пока не работает. Тоже потом допишу. К следующему НГ...

**Reply by Shark2805** [reply]
Огромное спасибо! Всё получилось! Ошибка была в настройках Jinx, прочитал всё внимательно и всё заработало. По вашей рекомендации воспользовался JinxFramer, ваши подозрения по поводу размеров оказались правильными, использовал их как отправную точку в поисках ошибки. Ёще раз спасибо и С Наступающим НН!
